compiler/rustc_builtin_macros/src/compile_error.rs

```diff
@@ -18,7 +18,7 @@ pub fn expand_compile_error<'cx>(
         reason = "diagnostic message is specified by user"
     )]
     #[expect(rustc::untranslatable_diagnostic, reason = "diagnostic message is specified by user")]
-    cx.span_err(sp, var.as_str());
+    cx.span_err(sp, var.as_str().to_string());
```

Comment:
does `Symbol` impl `ToString`?

Reply:
Good catch, it does. I will change this and similar ones to just `var.to_string()`.